neo3/wallet/wallet.py

```diff
@@ -0,0 +1,44 @@
+import json
+import os.path
+from .scrypt_parameters import ScryptParameters
+
+
+def encode_scrypt_parameters(scrypt_parameters):
+    if isinstance(scrypt_parameters, ScryptParameters):
+        return {'n': scrypt_parameters.n, 'r': scrypt_parameters.r, 'p': scrypt_parameters.p}
+    else:
+        type_name = scrypt_parameters.__class__.__name__
+        raise TypeError(f"Object of type '{type_name}' is not JSON serializable")
+
+
+class Wallet:
+
+    def __init__(self, path=None, name=None, json_wallet=None):
```

Comment:
Instead of having optionals in the "constructor" as a way of allowing different ways of constructing, please choose the most likely used as the default and make alternative constructors using classmethods. For example;

```python
T = TypeVar('T', bound='Wallet')

def __init__(self, 
             name: str, 
             version: str, 
             scrypt: ScryptParameters, 
             accounts: List[Account], 
             extra: Dict[any, any]):
   pass

@classmethod
def default(cls: Type[T]) -> T:
   return cls("", "3.0", ScryptParameters.default(), [], None)

@classmethod
def from_file(cls: Type[T], path: str) -> T:
   # parse data from file
   return from_json(data)
```